File: javascript/hamming/hamming.js
'use strict';

function Hamming() {}

Hamming.prototype.compute = function (strandA, strandB) {

    if (strandA.length != strandB.length) {
        throw new Error('DNA strands must be of equal length.');
    }

    var difference = 0;

    for (var i = 0; i < strandA.length; i++) {
        if (strandA[i] != strandB[i]) {
            difference++;
        };
    };

    return difference;
}

module.exports = Hamming;

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(strandA, strandB)
	local distance = 0
	for i = 1, string.len(strandA) do
		if string.sub(strandA, i, i) ~= string.sub(strandB, i, i) then
			distance = distance + 1
		end
	end
	return distance
end

return Hamming
